lib/chef/knife/bmcs_server_list.rb

```diff
@@ -25,10 +27,31 @@ def run
         options = {}
         options[:limit] = config[:limit] if config[:limit]
 
-        response = compute_client.list_instances(compartment_id, options)
+        servers = {}
 
-        display_list(response, ['Display Name', 'ID', 'State']) do |item|
-          [item.display_name, item.id, item.lifecycle_state]
+        instances = compute_client.list_instances(compartment_id, options)
+
+        # build a hash of hashs, keyed off of server_id
+        vnics = compute_client.list_vnic_attachments(compartment_id, {})
+        vnics.data.each do |vnic|
+          servers[vnic.instance_id] = {}
+          servers[vnic.instance_id]['instance_id'] = vnic.instance_id
+          servers[vnic.instance_id]['vnic_id'] = vnic.vnic_id
+        end
+
+        servers.each do |server, data|
+          vnic = network_client.get_vnic(data['vnic_id'], {})
```

Comment:
For an instance that is in the process of provisioning, I'm seeing this return a 404. You should catch that here. Also should check with terminated instances, and instances that don't have IP addresses (which can be created through the consoled UI, but not currently through the plugin).